packages/core/src/collaboration.ts

```diff
@@ -13,175 +13,41 @@
-
 /** @public */
-export function collaborativeDocFormat ({
-  documentId,
-  versionId,
-  lastVersionId,
-  source
-}: CollaborativeDocData): CollaborativeDoc {
-  const parts = [sanitize(documentId), sanitize(versionId), sanitize(lastVersionId)]
-  const collaborativeDoc = parts.join(':') as CollaborativeDoc
-  return collaborativeDocChain(collaborativeDoc, ...(source ?? []))
-}
-
-/**
- * Updates versionId component in the collaborative document.
- * Both versionId and lastVersionId will refer to the same collaborative document version.
- *
- * When versionId is not HEAD, the document will represent a readonly document version (snapshot).
- *
- * @public
- */
-export function collaborativeDocWithVersion (collaborativeDoc: CollaborativeDoc, versionId: string): CollaborativeDoc {
-  const { documentId, source } = collaborativeDocParse(collaborativeDoc)
-  return collaborativeDocFormat({ documentId, versionId, lastVersionId: versionId, source })
-}
-
-/**
- * Updates lastVersionId component in the collaborative document.
- *
- * When document versionId is HEAD, the function is no-op.
- *
- * @public
- */
-export function collaborativeDocWithLastVersion (
-  collaborativeDoc: CollaborativeDoc,
-  lastVersionId: string
-): CollaborativeDoc {
-  const { documentId, versionId, source } = collaborativeDocParse(collaborativeDoc)
-  return versionId === CollaborativeDocVersionHead
-    ? collaborativeDocFormat({ documentId, versionId, lastVersionId, source })
-    : collaborativeDoc
-}
-
-/**
- * Replaces source component in the collaborative document.
- *
- * @public
- */
-export function collaborativeDocWithSource (
-  collaborativeDoc: CollaborativeDoc,
-  source: CollaborativeDoc
-): CollaborativeDoc {
-  const { documentId, versionId, lastVersionId } = collaborativeDocParse(collaborativeDoc)
-  return collaborativeDocFormat({ documentId, versionId, lastVersionId, source: [source] })
-}
-
-/**
- * Creates collaborative document that refers to the last version from the source collaborative document.
- *
- * @public
- */
-export function collaborativeDocFromLastVersion (collaborativeDoc: CollaborativeDoc): CollaborativeDoc {
-  const { documentId, lastVersionId, source } = collaborativeDocParse(collaborativeDoc)
-  return collaborativeDocFormat({
-    documentId,
-    versionId: lastVersionId,
-    lastVersionId,
-    source
-  })
-}
-
-/**
- * Creates collaborative document that refers to the last version from the source collaborative document.
- *
- * @public
- */
-export function collaborativeDocFromCollaborativeDoc (
-  collaborativeDoc: CollaborativeDoc,
-  sourceCollaborativeDoc: CollaborativeDoc
-): CollaborativeDoc {
-  return collaborativeDocWithSource(collaborativeDoc, collaborativeDocFromLastVersion(sourceCollaborativeDoc))
+export function makeCollabJsonId (doc: CollaborativeDoc): Ref<Blob> {
+  const timestamp = Date.now()
+  const { objectId, objectAttr } = doc
+  return [objectId, objectAttr, timestamp].join('-') as Ref<Blob>
```

Comment:
Minor: are you parsing it back anywhere? If yes, `-` seems a very popular character that can easily be a part of `objectId` somewhere or sometime later.

Reply:
nope, it is for file name, it will never be parsed back. We could use any other character like % or |, but in file names `-` is much better because it does not require URL sanitization and visually looks better than other characters.

Reply:
Ok, let's keep it this way.